### DISCUSSION TITLE
Adding a bsdf material to a mitsuba.Mesh in the scene

### DISCUSSION THREAD

**Discussion by h-OUS-e** [mitsuba-renderer/mitsuba3#638] [Q&A] [answered]
Hello,

If we are creating a mesh from vertices and faces using `mi.Mesh()`, is there a way we can add a bsdf material? I am currently saving the mesh as a ".ply", and then reloading it with `mi.load_dict()` and setting `type=ply`, and adding a `bsdf` there, but it would be helpful if I don't have to save it and reload it to add a bsdf. 

Thanks!

**Comment by njroussel** [comment, marked as answer]
Hi @h-OUS-e 

You might want to take a look at this thread: https://github.com/mitsuba-renderer/mitsuba3/issues/109.

**Reply by h-OUS-e** [reply]
Ahh I searched discussions cuz I didn't think it was an issue, but exactl what I was looking for, thank you!
